bin/bst-speak.ts

```diff
@@ -30,62 +27,50 @@ program
         }
 
         // Just by casting program to options, we can get all the options which are set on it
-        let options: any = program;
-        let url = options.url;
-        let intentSchemaPath = options.intents;
-        let samplesPath = options.samples;
-        let applicationID = options.appId;
+        const options: any = program;
+        const token = options.token;
 
-        if (options.url === undefined) {
-            let proxyProcess = Global.running();
-            if (proxyProcess === null) {
-                console.error("No URL specified and no proxy is currently running");
+        let silentEchoResponse;
+        try {
+            silentEchoResponse = await SilentEchoClient.speak(utterance, token);
+        } catch (error) {
+            if (error.message === "Token Required") {
+                console.log("You need a token for this option to work, get it here:");
                 console.log();
-                console.log("URL (--url) must be specified if no proxy is currently running");
+                console.log("\thttps://apps.bespoken.io/dashboard/validation");
+                console.log("\t(Bespoken Dashboard account required for use of this feature)");
                 console.log();
-                console.log("If a proxy is running, utterances will automatically be sent to it");
+                console.log("Then try again with:");
+                console.log();
+                console.log("\tbst speak --token <ProvidedToken> <Speech to try>");
                 console.log();
                 process.exit(0);
                 return;
             }
-
-            url = "http://localhost:" + proxyProcess.port;
+            // Different issue, we throw to be catch by the general error handler.
+            throw error;
         }
 
-        let speaker = new BSTAlexa(url, intentSchemaPath, samplesPath, applicationID);
+        if (token) {
+            console.log("Your token is saved, you can now use this command without providing a token");
+        }
 
-        speaker.start(function (error: string) {
-            if (error !== undefined) {
-                process.exit(0);
-                return;
-            }
+        console.log("Transcript:");
```

Comment:
I would move the logic for calling SilentEcho and formatting the output to its own class. For two reasons:
1) The logic is a bit more complex, with lots of ifs (as almost anything can be null)
2) We will keep adding to it, with render templates, video, etc.
These additional may even require us creating HTML output, so the user can view it.

For that reason, having this handling in its own class makes sense